packages/launchpad/src/lcdapi/lcdclient.spec.ts

```diff
@@ -47,51 +35,39 @@ describe("LcdClient", () => {
   });
 
   describe("withModules", () => {
-    interface CodeInfo {
-      readonly id: number;
-      /** Bech32 account address */
-      readonly creator: string;
-      /** Hex-encoded sha256 hash of the code stored here */
-      readonly data_hash: string;
-      readonly source?: string;
-      readonly builder?: string;
-    }
-
-    type WasmResponse<T> = WasmSuccess<T> | WasmError;
-
-    interface WasmSuccess<T> {
+    interface TotalSupplyAllResponse {
       readonly height: string;
-      readonly result: T;
+      readonly result: LcdApiArray<Coin>;
     }
 
-    interface WasmError {
-      readonly error: string;
-    }
-
-    function isWasmError<T>(resp: WasmResponse<T>): resp is WasmError {
-      return (resp as WasmError).error !== undefined;
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    function setupSupplyExtension(base: LcdClient) {
+      return {
+        supply: {
+          totalAll: async (): Promise<TotalSupplyAllResponse> => {
+            return base.get(`/supply/total`);
+          },
+        },
+      };
     }
 
-    function unwrapWasmResponse<T>(response: WasmResponse<T>): T {
-      if (isWasmError(response)) {
-        throw new Error(response.error);
-      }
-      return response.result;
+    interface BankBalancesResponse {
+      readonly height: string;
+      readonly result: readonly Coin[];
     }
 
-    interface WasmExtension {
-      wasm: {
-        listCodeInfo: () => Promise<readonly CodeInfo[]>;
+    interface BankExtension {
+      readonly bank: {
+        readonly balances: (address: string) => Promise<BankBalancesResponse>;
       };
     }
 
-    function setupWasmExtension(base: LcdClient): WasmExtension {
+    function setupBankExtension(base: LcdClient): BankExtension {
```

Comment:
Why is this in the test code?
Don't we have such an implementation in the non-test (production) code?

Reply:
We have, but we need to reduce dependencies. Otherwise lcdclient depends on bank extension and bank extension depends on lcdclient. Having a wasm dependency here before was a huge upgrade pain.